packages/calypso-analytics/package.json

```diff
@@ -0,0 +1,43 @@
+{
+	"name": "@automattic/calypso-analytics",
+	"version": "1.0.0-alpha.0",
+	"description": "Automattic Analytics",
+	"homepage": "https://github.com/Automattic/wp-calypso",
+	"license": "GPL-2.0-or-later",
+	"author": "Automattic Inc.",
+	"sideEffects": false,
```

Comment:
This isn't accurate. At the very least `page-view-params` adds an event listener, `tracks` adds `_tkq` to the window object. I haven't audited the rest of the modules, but this should be fixed.

Reply:
@roo2 marked as resolved but this still doesn't look right to me. Unless I've misunderstood something, we can't mark this package as `sideEffects=false` because importing code from this package does indeed have side-effects (e.g. the event listener).
Does this need to be `sideEffects=true`?

Reply:
Ahh sorry I accidentally hit mark as resolve instead of posting my comment here 🤐. 
It looks like `sideEffects` is used to determine if webpack is allowed to do tree shaking and not load files that aren't needed. If the module has `sideEffects` it means that there are implicit dependencies between functions. So "side effects" like attaching the event handler to window don't count as `sideEffects`, as long as other functions don't depend on it and it could safely be left out by the tree shaking build. 
https://stackoverflow.com/questions/49160752/what-does-webpack-4-expect-from-a-package-with-sideeffects-false

We used to have a cross function dependency on `window._tkq` being initialized at the top of the file, but I've refactored it so that `window._tkq` is not used directly, rather, the wrapping function `pushEventToTracksQueue` is used. I've also updated other places in calypso (`client/lib/analytics/index.js`) to use `pushEventToTracksQueue`. This way there are no implicit dependencies (despite still using the global) and the tree shaker can trace down to include only the functions that it needs.

So yes I believe it is correct to set `sideEffects:false` now, despite the code having "side effects".

Reply:
> "side effects" like attaching the event handler to window don't count as sideEffects, as long as other functions don't depend on it and it could safely be left out by the tree shaking build

Hmmm I hadn't thought of it like that 🤔

I guess in theory it shouldn't make a difference whether there's a latent window listener there or not. But I'm worried it's the sort of thing that'd be really hard to track down if it _did_ cause a difference between dev and prod builds. I'm thinking of cases where it was tricky fixing a Renovate PR because the modules getting updated had side effects.

It's not what I'd expect from something that's labelled side-effect free, but I see from the SO post that there's a fuzzy line since things like `console.log`s are generally ok. Do you have any opinions on where to draw the line @jon?

Another way would be to add the event listener in `initializeAnalytics()`, but that'd only work if it was being called early enough.

Reply:
hmm It looks like setting `sideEffects=true` might be a safer and clearer option and I think that this module is so small and all in use that tree shaking wouldn't really make a difference, I'll just set it to true I think.

Reply:
Sweet, you did bring up a good point though. I'm interested in figuring out where we should be drawing that line in general.

Reply:
You can also set it to a sideEffects to an array of files that have sideEffects, but just `true` is probably enough here. We aren't likely to make big gains here.